### PR TITLE
Add zap logger and use snarkjs proof result directly

### PR DESCRIPTION
**Pull request by vmidyllic**
1.  Instead of using the go-circom-prover-verifier library,  here we used plain results from the snark js library
2. Add context zap logger

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
+service:
+  golangci-lint-version: 1.39.x
```
Comment (Kolezhniuk) [inline .golangci.yml]: increase version?

**Review by Kolezhniuk — APPROVED** [review]

File: .golangci.yml (hunk 1 above)
Comment (vmidyllic) [reply .golangci.yml]: upgraded

File: pkg/app/middleware/logger.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		ww := middleware.NewWrapResponseWriter(w, r.ProtoMajor)
+
+		t1 := time.Now()
+		defer func() {
+			log.WithContext(r.Context()).Infow("http	",
+				"method", r.Method,
+				"path", r.URL.Path,
+				"remoteAddr", r.RemoteAddr,
+				"lat", time.Since(t1),
```
Comment (OBrezhniev) [inline pkg/app/middleware/logger.go]: I would suggest to rename "lat" to "responseTime" or "duration", and have it in milliseconds

**Review by OBrezhniev — CHANGES_REQUESTED** [review]

File: pkg/app/server.go [hunk 3]
```diff
@@ -3,22 +3,25 @@ package app
...
 	}
 }
 
+// Run starts the server
 func (s *Server) Run(port int) {
-	log.Infof("Server starting on port %d", port)
+	log.Infow("Server started", "port", port)
 	err := http.ListenAndServe(fmt.Sprintf(":%d", port), s.Routes)
-	log.Fatal(err)
+	log.Panic("server stopped", err)
```
Comment (OBrezhniev) [inline pkg/app/server.go]: From the zap dicumentation:

> Panic uses fmt.Sprint to construct and log a message, then panics.

Probably panicking is not needed here.

**Review by OBrezhniev — CHANGES_REQUESTED** [review]

File: pkg/app/middleware/logger.go (hunk 2 above)
Comment (vmidyllic) [reply pkg/app/middleware/logger.go]: fixed

File: pkg/app/server.go (hunk 3 above)
Comment (vmidyllic) [reply pkg/app/server.go]: fixed

**Review by OBrezhniev — APPROVED** [review]

**Pull request merged**